### PR TITLE
Issue #32981 || Corrected Spinbutton and Slider role "only read-write" content and did some changes on both so that both roles can be more consistent and similar.

### PR DESCRIPTION
**Pull request by SASHWAT12**
…e additional changes for both the roles to be more consistent and similar

### CODE REVIEW

File: files/en-us/web/accessibility/aria/roles/slider_role/index.md [hunk 1]
```diff
@@ -15,15 +15,15 @@ The `slider` role is for range input widgets where the user selects a value from
 
 ### The `slider` role compared other range options
 
-ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, and sliders.
+ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, scrollbars, range input, spinbuttons and sliders.
 
-The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range indicating the portion of completion of a task, progressing in a single direction, such as a file upload's loading progress bar that eventually reaches 100% when fully loaded.
+The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range. It indicates the portion of completion of a task, progressing in a single direction, such as a file upload's loading progress bar that eventually reaches 100% when fully loaded.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/slider_role/index.md:20]:
```suggestion
The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range. It indicates the portion of completion of a task, progressing in a single direction. For instance, a file upload's progress bar that conveys the data transfer up to 100%, when the file has fully uploaded.
```

File: files/en-us/web/accessibility/aria/roles/slider_role/index.md [hunk 2]
```diff
@@ -15,15 +15,15 @@ The `slider` role is for range input widgets where the user selects a value from
...
 
-The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only gauge indicating the amount of something within a known range, such as a computer's battery indicator or a car's gas gauge.
+The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only gauge. It indicates the amount of something within a known range, such as a computer's battery indicator or a car's gas gauge.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/slider_role/index.md:22]:
```suggestion
The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only range. It represents a scalar measurement within a known range, or a fractional value. For instance, a computer's battery indicator, the disk usage of a hard drive or a car's gas gauge.
```

File: files/en-us/web/accessibility/aria/roles/slider_role/index.md [hunk 3]
```diff
@@ -15,15 +15,15 @@ The `slider` role is for range input widgets where the user selects a value from
...
+The `slider` role, similar to HTML's `input` of type `range`, [`<input type="range">`](/en-US/docs/Web/HTML/Element/input/range), is a read-write input range. Sliders allow users to select a value between predefined minimum and maximum values. The user selects a value by moving a slider thumb along a horizontal or vertical slider to select a value.
 
-While all three of these ranges have the same ARIA states and properties, the `slider` role is the only read-write range: it is the only one whose value changes via user interaction. As such, it must be able to receive focus. In addition, keyboard interaction, mouse clicks, and touch interaction must be supported.
+The `slider` role is a read-write range: it's value changes via user interaction. As such, it must be able to receive focus. In addition, keyboard interaction, mouse clicks, and touch interaction must be supported.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/slider_role/index.md:26]: it was already mentioned that the slider is a read-write input range, so it doesn't need to be repeated.  

```suggestion
A `slider` role's value changes via user interaction. As such, it must be able to receive focus. In addition, to adjust the value keyboard interaction, mouse clicks, and touch interaction must be supported.
```

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md [hunk 4]
```diff
@@ -24,15 +24,15 @@ In addition to including the [`tabindex`](/en-US/docs/Web/HTML/Global_attributes
 
 ### ARIA range widget options
 
-ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, sliders and spinbuttons.
+ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, scrollbars, range input, sliders and spinbuttons.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:27]: again a range input is not an aria role.  a range input (html) is the same as a role=slider.

```suggestion
ARIA provides developers with five different [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles) that inherit from the range abstract role, including progressbars, meters, scrollbars, spinbuttons and sliders.
```

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md [hunk 5]
```diff
@@ -24,15 +24,15 @@ In addition to including the [`tabindex`](/en-US/docs/Web/HTML/Global_attributes
...
 
 The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range. It indicates the portion of completion of a task, progressing in a single direction, such as a file upload's loading progress bar that eventually reaches 100% when fully loaded.
 
 The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only gauge. It indicates the amount of something within a known range, such as a computer's battery indicator or a car's gas gauge.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:31]:
```suggestion
The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only range. It represents a scalar measurement within a known range, or a fractional value. For instance, a computer's battery indicator, the disk usage of a hard drive or a car's gas gauge.
```

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md [hunk 6]
```diff
@@ -24,15 +24,15 @@ In addition to including the [`tabindex`](/en-US/docs/Web/HTML/Global_attributes
 
 ### ARIA range widget options
 
-ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, sliders and spinbuttons.
+ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, scrollbars, range input, sliders and spinbuttons.
 
 The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range. It indicates the portion of completion of a task, progressing in a single direction, such as a file upload's loading progress bar that eventually reaches 100% when fully loaded.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:29]:
```suggestion
The [`progressbar`](/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) role, similar to HTML's {{HTMLElement('progress')}} element, is a read-only range. It indicates the portion of completion of a task, progressing in a single direction. For instance, a file upload's progress bar that conveys the data transfer up to 100%, when the file has fully uploaded.
```

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md [hunk 7]
```diff
@@ -24,15 +24,15 @@ In addition to including the [`tabindex`](/en-US/docs/Web/HTML/Global_attributes
...
 
 The [`meter`](/en-US/docs/Web/Accessibility/ARIA/Roles/meter_role) role, similar to HTML's {{HTMLElement('meter')}} element, is a read-only gauge. It indicates the amount of something within a known range, such as a computer's battery indicator or a car's gas gauge.
 
 The `slider` role, similar to HTML's `input` of type `range`, [`<input type="range">`](/en-US/docs/Web/HTML/Element/input/range), is a read-write input range. Sliders allow users to select a value between the predefined minimum and maximum values. The user selects a value by moving a slider thumb along a horizontal or vertical slider to select a value.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:33]:
```suggestion
The `slider` role, similar to HTML's `input` of type `range`, [`<input type="range">`](/en-US/docs/Web/HTML/Element/input/range), is a read-write input range. Sliders allow users to select a value between predefined minimum and maximum values. The user selects a value by moving a slider thumb along a horizontal or vertical slider to select a value.
```

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md [hunk 8]
```diff
@@ -24,15 +24,15 @@ In addition to including the [`tabindex`](/en-US/docs/Web/HTML/Global_attributes
...
 
-While all three of these ranges have the same ARIA states and properties, the `spinbutton` role is the only read-write range: it is the only one whose value changes via user interaction. As such, it must be able to receive focus. In addition, keyboard interaction, mouse clicks, and touch interaction must be supported.
+The `spinbutton` role is a read-write range: it's value changes via user interaction. As such, it must be able to receive focus. In addition, keyboard interaction, mouse clicks, and touch interaction must be supported.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:35]: this seems like it was pulled from the slider doc and just replaced slider with spinbutton.  The deletion of saying it was the only one whose value changes via user interaction was a good fix, but might as well call out the similarities between slider and spinbutton here.

```suggestion
The `spinbutton` and `slider` roles' values change via user interaction. As such, these roles must be able to receive focus. In addition, to adjust the value keyboard interaction, mouse clicks, and touch interaction must be supported.
```

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md [hunk 9]
```diff
@@ -24,15 +24,15 @@ In addition to including the [`tabindex`](/en-US/docs/Web/HTML/Global_attributes
...
 
 > [!WARNING]
 > To change the spinbutton value, touch-based assistive technologies need to respond to user gestures for increasing and decreasing the value by synthesizing key events.
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md:38]: This warning seems like it was copied over from the slider doc. It's not really accurate/applicable to most spinbutton implementations, so i suggest removing it.

```suggestion
```

**Review by scottaohara — CHANGES_REQUESTED** [review]
good first pass in trying to fix up these two docs.

i've taken it a bit further though, and corrected some inaccuracies with the original PR proposal.

File: files/en-us/web/accessibility/aria/roles/slider_role/index.md [hunk 10]
```diff
@@ -15,15 +15,15 @@ The `slider` role is for range input widgets where the user selects a value from
 
 ### The `slider` role compared other range options
 
-ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, and sliders.
+ARIA provides developers with six different range [widget roles](/en-US/docs/Web/Accessibility/ARIA/Roles#2._widget_roles), including progressbars, meters, scrollbars, range input, spinbuttons and sliders.
```
[the remark this answers is not in the document]
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/slider_role/index.md]: Thank you for highlighting the inaccuracy. I have implemented the change.

File: files/en-us/web/accessibility/aria/roles/slider_role/index.md (hunk 1 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/slider_role/index.md]: I have implemented the change.

File: files/en-us/web/accessibility/aria/roles/slider_role/index.md (hunk 3 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/slider_role/index.md]: Thanks for highlighting, repetitive stuff is corrected.

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md (hunk 4 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md]: Same changes are implemented

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md (hunk 5 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md]: Suggested changes are implemented.

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md (hunk 6 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md]: Requested changes are implemented

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md (hunk 7 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md]: Changes are made

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md (hunk 8 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md]: I'm glad you found the fix helpful. I understood your point about highlighting the similarities between the slider and spinbutton and the changes has been incorporated.

File: files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md (hunk 9 above)
Comment (SASHWAT12) [reply files/en-us/web/accessibility/aria/roles/spinbutton_role/index.md]: As suggested the less applicable warning is removed. Please let me know if any further improvements can be done.

**Comment by SASHWAT12** [comment]
Implemented requested changes from review, further making content more accurate and clearer.  Please let me know if any further improvements can be done.

**Comment by SASHWAT12** [comment]
> good first pass in trying to fix up these two docs.
> 
> i've taken it a bit further though, and corrected some inaccuracies with the original PR proposal.

**Pull request closed without merging**